soda/core/soda/execution/data_source_scan.py

```diff
@@ -65,16 +71,24 @@ def execute_queries(self):
         for query in all_data_source_queries:
             query.execute()
 
-    def create_automated_monitor_run(self, automated_monitoring_cfg, scan):
-        from soda.execution.automated_monitoring_run import AutomatedMonitoringRun
+    def run(self, data_source_check_cfg: DataSourceScanCfg, scan: "Scan"):
+        if isinstance(data_source_check_cfg, AutomatedMonitoringCfg):
+            from soda.execution.automated_monitoring_run import AutomatedMonitoringRun
 
-        return AutomatedMonitoringRun(self, automated_monitoring_cfg)
+            automated_monitoring_run = AutomatedMonitoringRun(self, data_source_check_cfg).run()
+            scan._checks.extend(automated_monitoring_run)
 
-    def create_profile_columns_run(self, profile_columns_cfg, scan):
-        return ProfileColumnsRun(self, profile_columns_cfg)
+        if isinstance(data_source_check_cfg, ProfileColumnsCfg):
+            profile_columns_run = ProfileColumnsRun(self, data_source_check_cfg).run()
+            scan._profile_columns_result_tables.extend(profile_columns_run.tables)
+            scan._is_profiling_run = True  # TODO: delete this line
 
-    def create_discover_tables_run(self, data_source_check_cfg, scan):
-        return DiscoverTablesRun(self, data_source_check_cfg)
+        if isinstance(data_source_check_cfg, DiscoverTablesCfg):
+            discover_tables_run = DiscoverTablesRun(self, data_source_check_cfg).run()
+            scan._discover_tables_result_tables.extend(discover_tables_run.tables)
+            scan._is_profiling_run = True  # TODO: delete this line
```

Comment:
What happens when you remove this?

Reply:
the unit tests will fail in the current code. If we merge your [PR](https://github.com/sodadata/soda-core/pull/1377), then I'll delete this line

Reply:
What in my PR helps? The test assertion helper?